README.md
# Tentang Bash dalam Bahasa Indonesia

Project bash-in-bahasa adalah project for fun untuk menterjemahkan beberapa kata kunci dalam Bash shell kedalam Bahasa Indonesia. Sehingga ketika kita berinteraksi dengan shell atau menulis shell script kita dapat menggunakan kata yang ada dalam Bahasa Indonesia.

* [Penggunaan](#penggunaan)
* [Contoh-contoh](#contoh-contoh)
* [Kekurangan](#kekurangan)
* [Kontribusi Bahasa Lain](#kontribusi-bahasa-lain)
* [Penulis](#penulis)
* [Lisensi](#lisensi)

## Penggunaan

Untuk menjalankan cukup impor file bash-in-bahasa.sh dan Bash dalam Bahasa Indonesia siap digunakan.

```
$ git clone git@github.com:rioastamal/bash-in-bahasa.git
$ cd bash-in-bahasa
```

Kita dapat menggunakan perintah source atau `.` titik untuk impor file.

```
$ source bash-in-bahasa.sh
$ tulis "Halo Dunia"
Halo Dunia
```

Contoh lain.

```
$ untuk (( i=1; i<=5; i++ ))
> lakukan
>   tulis "Baris: ${i}"
> selesai
Baris: 1
Baris: 2
Baris: 3
Baris: 4
Baris: 5
```

Jika menggunakan bash-in-bahasa pada Shell script kamu perlu menggunakan opsi `shopt -s expand_aliases` agar alias bash-in-bahasa digunakan.

```
#!/bin/bash
shopt -s expand_aliases

# Panggil alias bash-in-bahasa
source ./bash-in-bahasa.sh

# Alias Bahasa Indonesia sekarang bisa digunakan
tulis 'Hello World'
```

## Contoh-contoh

Kamu dapat melihat beberapa contoh yang sudah dibuat pada direktori `samples/`. Ini adalah contoh game tebak angka sederhana yang disertakan dalam kode sumber repo ini.

```bash
#!/bin/bash
#
# Contoh file penggunaan alias bash-in-bahasa.

# Perintah ini wajib disertakan agar alias dikenali
shopt -s expand_aliases

# Asumsi anda menjalankan sample script ini dari root direktori bash-in-bahasa
# $ bash samples/tebak-angka.sh
. ./bash-in-bahasa.sh

maksimal_percobaan=3
angka_minimum=1
angka_maksimum=15
jawaban_pengguna="salah"
percobaan=1
# Lihat https://stackoverflow.com/questions/8988824/generating-random-number-between-1-and-10-in-bash-shell-script
jawaban=$(( ($RANDOM % $angka_maksimum) + 1 ))

tulis '**********************************'
tulis 'SELAMAT DATANG DI GAME TEBAK ANGKA'
tulis '**********************************'
tulis "Kamu harus menebak angka yang muncul maksimal ${maksimal_percobaan} kali percobaan.
Angka berada diantara ${angka_minimum} dan ${angka_maksimum}. Selamat bermain!"

tulis
tulis -n '> Masukkan nama kamu: '
baca_input nama_pengguna

tulis

selama (( $percobaan <= $maksimal_percobaan ))
lakukan
    tulis -n "> #${percobaan} Masukkan angka: "
    baca_input jawaban_pengguna

    jika [ $jawaban = $jawaban_pengguna ];
    maka
        tulis "Selamat ${nama_pengguna}, jawaban kamu BENAR!"
        keluar 0
    akhir_jika

    percobaan=$(( $percobaan + 1 ))
selesai

tulis "GAME OVER. Ooops kamu sudah melakukan ${maksimal_percobaan} kali percobaan."
keluar 1
```

Cara menjalankannya.

```
$ bash samples/tebak-angka.sh
**********************************
SELAMAT DATANG DI GAME TEBAK ANGKA
**********************************
Kamu harus menebak angka yang muncul maksimal 3 kali percobaan.
Angka berada diantara 1 dan 15. Selamat bermain!

> Masukkan nama kamu: Rio

> #1 Masukkan angka: 3
> #2 Masukkan angka: 6
> #3 Masukkan angka: 1
GAME OVER. Ooops kamu sudah melakukan 3 kali percobaan.
```

## Kekurangan

Terdapat kekurangan dalam bash-in-bahasa yaitu tidak berfungsinya perulangan `for...in...`. Harusnya alias tersebut dapat ditulis `untuk...dalam...` tetapi entah kenapa Bash tidak bisa melakukan interpretasi.

```
$ source bash-in-bahasa.sh
$ untuk i dalam 1 2 3
-bash: syntax error near unexpected token `dalam'
```

## Kontribusi Bahasa Lain

Penulis terbuka untuk membuka kontribusi dalam bahasa lain misal `bash-in-jawa` atau `bash-in-sunda` dan lain-lain. Silahkan submit PR bahasa alias baru tersebut sesuai dengan contoh yang ada di `bahs-in-bahasa.sh`.

## Penulis

bash-in-bahasa dibuat oleh Rio Astamal &lt;rio@rioastamal.net&gt;

## Lisensi

Project ini dilisensikan dibawah naungan MIT License. Lihat file LICENSE.md.


CHANGELOG.md
## Release v1.1 - 28 January 2020

- New functions for cursor movement
- New sample file: hangman.sh

## Release v1.0 - 13 January 2020

- Initial release
- Subtitute most of Bash keywords to Bahasa Indonesia
- Sample files: piramida.sh and tebak-angka.sh

bash-in-bahasa.sh
#!/bin/bash
#
# List of alias of Bash keywords in Bahasa Indonesia
#
# @author Rio Astamal <rio@rioastamal.net>
# @license MIT

# Generated using `compgen -k`
alias jika=if
alias jk=if
alias maka=then
alias selain_itu=else
alias seltu=else
alias selain_itu_jika=elif
alias seltu_jika=elif
alias kaji=fi
alias akhir_jika=fi
alias kj=fi
alias kasus=case
alias hal=case
alias susak=esac
alias lah=esac
alias untuk=for
alias pilih=select
alias selama=while
alias sampai=until
alias lakukan=do
alias selesai=done
alias dalam=in
alias fungsi=function
alias waktu=time

# Generated using `compgen -b`
alias namalain=alias
alias proses_latar=bg
alias mengikat=bind
alias putus=break
alias berhenti=break
#builtin
alias pemanggil=caller
alias masuk=cd
alias perintah=command
#compgen
alias lengkap=complete
alias teruskan=continue
alias deklarasikan=declare
#dirs
alias pungkiri=disown
alias cetak=echo
alias tulis=echo
alias izinkan=enable
alias evaluasi=eval
alias eksekusi=exec
alias keluar=exit
#exporexport
alias salah=false
#fc
alias proses_depan=fg
alias ambilopsi=getopts
# hash
alias bantuan=help
alias riwayat=history
alias kerjaan=jobs
alias bunuh=kill
alias biarkan=let
alias lokal=local
# logout
# popd
alias cetakf=printf
# pushd
alias dir_sekarang=pwd
alias baca=read
alias baca_input=read
alias bacasaja=readonly
alias kembalikan=return
alias nilai_balik=return
alias tetapkan=set
alias geser_arg=shift
# shopt
alias sumber=source
alias tangguhkan=suspend
alias coba=test
alias akumulasi_waktu=times
alias jebakan=trap
alias benar=true
alias tipe=type
#typeset
#ulimit
#umask
alias hilangkan_namalain=unalias
alias jangan_tetapkan=unset
alias tunggu=wait

kursor_ke_atas()
{
    tulis -ne "\033[${1}A"
}

kursor_ke_bawah()
{
    tulis -ne "\033[${1}B"
}

kursor_ke_kanan()
{
    tulis -ne "\033[${1}C"
}

kursor_ke_kiri()
{
    tulis -ne "\033[${1}D"
}

bersihkan_layar()
{
    tulis -ne ""
}

bersihkan_baris()
{
    tulis -ne "\033[K"
}

tulis_bersih()
{
    tulis -ne "${1}\033[K"
}

letakkan_kursor()
{
    tulis -ne "\033[${1};${2}H"
}

samples/hangman.sh
#!/bin/bash
#
# Hangman: sebuah permainan tebak kata sederhana dimana pemain harus
# menebak kata dengan benar dengan menyebutkan huruf satu per satu.
#
# Jika gagal maka gambar orang digantung (hangman) akan dimunculkan
shopt -s expand_aliases

DIR_SEKARANG=$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )
source "$DIR_SEKARANG/../bash-in-bahasa.sh"

gambar_tulisan_hangman()
{
    tulis '##     ##    ###    ##    ##  ######   ##     ##    ###    ##    ##'
    tulis '##     ##   ## ##   ###   ## ##    ##  ###   ###   ## ##   ###   ##'
    tulis '##     ##  ##   ##  ####  ## ##        #### ####  ##   ##  ####  ##'
    tulis '######### ##     ## ## ## ## ##   #### ## ### ## ##     ## ## ## ##'
    tulis '##     ## ######### ##  #### ##    ##  ##     ## ######### ##  ####'
    tulis '##     ## ##     ## ##   ### ##    ##  ##     ## ##     ## ##   ###'
    tulis '##     ## ##     ## ##    ##  ######   ##     ## ##     ## ##    ##'
    tulis
}

bersihkan_hangman()
{
    tulis_bersih ""
    untuk (( baris=1; baris<=7; baris++ )); lakukan
        tulis_bersih "\n"
    selesai
    kursor_ke_atas 7
}

gambar_hangman()
{
    bersihkan_hangman

    jika [ $1 = 0 ]; maka
        tulis -e "\n\n\n\n\n\n\n"
        kembalikan 0
    akhir_jika

    jika [ $1 = 1 ]; maka
        tulis -e "\n"
        tulis "                                    O"
        tulis "                                    |"
        tulis -e "\n\n\n"
        kembalikan 0
    akhir_jika

    jika [ $1 = 2 ]; maka
        tulis -e "\n"
        tulis "                                    O "
        tulis "                                   /|\ "
        tulis -e "\n\n\n"
        kembalikan 0
    akhir_jika

    jika [ $1 = 3 ]; maka
        tulis -e "\n"
        tulis "                                    O "
        tulis "                                   /|\ "
        tulis "                                   / \ "
        tulis -e "\n\n"
        kembalikan 0
    akhir_jika

    jika [ $1 = 4 ]; maka
        tulis "                                    + "
        tulis "                                    | "
        tulis "                                    O "
        tulis "                                   /|\ "
        tulis "                                   / \ "
        tulis -e "\n\n"
        kembalikan 0
    akhir_jika

    tulis "                          [|]=======+ "
    tulis "    +------+              [|]       | "
    tulis "    | GAME |              [|]       @ "
    tulis "    | OVER |              [|]      /|\ "
    tulis "    +------+              [|]      / \ "
    tulis "                          [|] "
    tulis "                          [|] "
    tulis "                          === "
}

gambar_kemenangan()
{
   bersihkan_hangman

    tulis "    +---------+  ''''''' "
    tulis "    | KAMU    |    \O/ "
    tulis "    | MENANG! |     | "
    tulis "    +---------+    / \ "
    tulis "                 -------"
    tulis ""
    tulis ""
}

tulis_garis_bawah()
{
    lokal jumlah_huruf=${#1}
    lokal huruf_dicari=
    lokal akhiran=" "

    jika [ "$2" = "kosong" ]; maka
        akhiran=""
    akhir_jika

    untuk (( garis=0; garis<$jumlah_huruf; garis++ ))
    lakukan
        jika [ "$huruf_benar" = "" ]; maka
            tulis -n "__${akhiran}"
            teruskan
        akhir_jika

        # Tes apakah huruf ada string jawaban
        # @lihat https://stackoverflow.com/questions/229551/how-to-check-if-a-string-contains-a-substring-in-bash
        huruf_dicari=${1:$garis:1}
        jika [ -z "${huruf_benar##*$huruf_dicari*}" ]; maka
            tulis -n "${huruf_dicari}${akhiran}"
            teruskan
        akhir_jika

        tulis -n "__${akhiran}"
    selesai
}

adakah_huruf_dalam()
{
    lokal kata_sumber=$1
    lokal huruf_dicari=$2

    jika [ -z "${kata_sumber##*$huruf_dicari*}" ]; maka
        tulis 'ya'
        kembalikan 0
    akhir_jika

    tulis 'tidak'
    kembalikan 1
}

kembalikan_nilai_awal()
{
    jawaban="$( acak_jawaban )"
    jawaban_pengguna=""
    huruf_benar=""
    percobaan=0
}

acak_jawaban()
{
    # Daftar negara dan ibu kota
    lokal daftar_jawaban=( AFGHANISTAN ALBANIA ALJAZAIR ANDORRA ANGOLA ANTIGUA-DAN-BARBUDA ARGENTINA ARMENIA AUSTRALIA AUSTRIA AZERBAIJAN BAHAMA BAHRAIN BANGLADESH BARBADOS BELARUS BELGIA BELIZE BENIN BHUTAN BOLIVIA BOSNIA-HERZEGOVINA BOTSWANA BRASIL BRUNEI-DARUSSALAM BULGARIA BURKINA-FASO BURUNDI KAMBOJA KAMERUN KANADA CAPE-VERDE AFRIKA-TENGAH CHAD CHILE CINA KOLOMBIA KOMORO KONGO KOSTA-RIKA PANTAI-GADING KROASIA KUBA SIPRUS REPUBLIK-CEKO DENMARK DJIBOUTI DOMINIKA REPUBLIK-DOMINIKA TIMOR-LESTE EKUADOR MESIR EL-SALVADOR GUINEA-EKUATORIAL ERITREA ESTONIA ETHIOPIA FIJI FINLANDIA PRANCIS GABON GAMBIA GEORGIA JERMAN GHANA YUNANI GRENADA GUATEMALA GUINEA GUINEA-BISSAU GUYANA HAITI HONDURAS HONGARIA ISLANDIA INDIA INDONESIA IRAN IRAK IRLANDIA ISRAEL ITALIA JAMAIKA JEPANG YORDANIA KAZAKHSTAN KENYA KIRIBATI KOREA-UTARA KOREA-SELATAN KUWAIT KYRGYZSTAN LAOS LATVIA LEBANON LIBYA LIECHTENSTEIN LUXEMBOURG MACEDONIA MADAGASKAR MALAWI MALAYSIA MALADEWA MALI MALTA KEPULAUAN-MARSHALL MAURITANIA MAURITIUS MEKSIKO FEDERASI-MIKRONESIA MOLDOVA MONAKO MONGOLIA MONTENEGRO MAROKO MOZAMBIK MYANMAR NAMIBIA NAURU NEPAL BELANDA SELANDIA-BARU NIKARAGUA NIGER NIGERIA NORWEGIA OMAN PAKISTAN PALAU PANAMA PAPUA-NUGINI PARAGUAY PERU FILIPINA POLANDIA PORTUGAL QATAR ROMANIA RUSIA RWANDA SAINT-KITTS-DAN-NEVIS SAINT-LUCIA SAINT-VINCENT-DAN -GRENADINES SAMOA SAN-MARINO SAO-TOME-DAN-PRINCIPE ARAB-SAUDI SENEGAL SERBIA SEYCHELLES SIERRA-LEONE SINGAPURA SLOVAKIA SLOVENIA KEPULAUAN-SOLOMON SOMALIA AFRIKA-SELATAN SPANYOL SRI-LANKA SUDAN SURINAME SWAZILAND SWEDIA SWISS SURIAH TAJIKISTAN TANZANIA THAILAND TOGO TONGA TRINIDAD-DAN-TOBAGO TUNISIA TURKI TURKMENISTAN TUVALU UGANDA UKRAINA UNI-EMIRAT-ARAB INGGRIS AMERIKA-SERIKAT URUGUAY UZBEKISTAN VANUATU VATIKAN VENEZUELA VIETNAM YAMAN ZAMBIA ZIMBABWE KABUL TIRANA ALJIR ANDORRA-LA-VELLA LUANDA SAINT-JOHNS BUENOS-AIRES YEREVAN CANBERRA WINA BAKU NASSAU MANAMA DHAKA BRIDGETOWN MINSK BRUSSEL BELMOPAN PORTO-NOVO THIMPHU LA-PAZ SARAJEVO GABORONE BRASILIA BANDAR-SERI-BEGAWAN SOFIA OUAGADOUGOU BUJUMBURA PHNOM-PENH YAOUNDE OTTAWA PRAIA BANGUI NDJAMENA SANTIAGO BEIJING BOGOTA MORONI BRAZZAVILLE SAN-JOSE ABIJAN ZAGREB HAVANA NIKOSIA PRAHA KOPENHAGEN DJIBOUTI ROSEAU SANTO-DOMINGO DILI QUITO KAIRO SAN-SALVADOR MALABO ASMARA TALLINN ADDIS-ABABA SUVA HELSINKI PARIS LIBREVILLE BANJUL TBILISI BERLIN AKRA ATHENA SAINT-GEORGE GUATEMALA-CITY CONAKRY BISSAU GEORGETOWN PORT-AU-PRINCE TEGUCIPALGA BUDAPEST REYKJAVIK NEW-DELHI JAKARTA TEHERAN BAGHDAD DUBLIN TEL-AVIV ROMA KINGSTON TOKYO AMMAN ASTANA NAIROBI TARAWA PYONGYANG SEOUL KUWAIT-CITY BISHKEK VIENTIANE RIGA BEIRUT TRIPOLI VADUZ LUXEMBOURG SKOPJE ANTANANARIVO LILONGWE KUALA-LUMPUR MALE BAMAKO MALTA MAJURO NOUAKCHOTT PORT-LOUIS MEXICO-CITY PALIKIR CHISINAU MONAKO ULAN-BATOR PODGORICA RABAT MAPUTO NAYPYIDAW WINDHOEK KATHMANDU AMSTERDAM WELLINGTON MANAGUA NIAMEY ABUJA OSLO MUSKAT ISLAMABAD KOROR PANAMA-CITY PORT-MORESBY ASUNCION LIMA MANILA WARSAWA LISABON DOHA BUCHAREST MOSKOW KIGALI BASSETERRE CASTRIES KINGSTOWN APIA SAN-MARINO SAO-TOME RIYADH DAKAR BEOGRAD VICTORIA FREETOWN SINGAPURA BRATISLAVA LJUBLJANA HONIARA MOGADISHU PRETORIA MADRID KOLOMBO KHARTOUM PARAMARIBO MBABANE STOCKHOLM BERN DAMASKUS DUSHANBE DAR-ES-SALAAM BANGKOK LOME NUKUALOFA PORT-OF-SPAIN TUNIS ANKARA ASHGABAT FUNAFUTI KAMPALA KIEV ABU-DHABI LONDON WASHINGTON-DC MONTEVIDEO TASHKENT PORT-VILA VATIKAN CARACAS HANOI SANAA LUSAKA HARARE )
    lokal total_jawaban=$( echo ${#daftar_jawaban[@]} )
    lokal indeks=$(( ($RANDOM % $total_jawaban) + 1 ))

    echo ${daftar_jawaban[$indeks]}
}

maksimal_percobaan=5
jawaban="$( acak_jawaban )"
jawaban_pengguna=""
huruf_benar=""
percobaan=0

gambar_tulisan_hangman

# Perulangan utama
selama benar :
lakukan
    tulis -ne "\r"
    gambar_hangman $percobaan

    tulis -e "Petunjuk: Nama negara atau ibu kota. Jika spasi ganti dengan tanda \"-\".\n"
    tulis_bersih ""
    tulis_garis_bawah "$jawaban"

    tulis -e  "\n"
    tulis_bersih "Masukkan satu huruf (KAPITAL): "
    tulis_bersih "\n"

    jika [ "$jawaban_pengguna" = "$jawaban" ]; maka
        kursor_ke_atas 13
        gambar_kemenangan
        kursor_ke_bawah 5
        tulis -ne "\n>> Tekan ENTER untuk permainan baru atau CTRL+C untuk keluar."

        baca_input huruf

        kembalikan_nilai_awal
        kursor_ke_atas 14
        teruskan
    akhir_jika

    jika (( $percobaan > 0 )); maka
        tulis_bersih "Gagal ke-$(( ${percobaan} )) dari ${maksimal_percobaan}."
    akhir_jika

    kursor_ke_atas 1

    # reset kursor ke paling kiri sebelum digeser kenanan
    tulis -ne "\r"
    kursor_ke_kanan 31

    jika (( $percobaan >= $maksimal_percobaan )); maka
        percobaan=0
        kursor_ke_atas 2
        tulis_bersih "\n\nKamu sudah ${maksimal_percobaan} kali GAGAL. Jawaban yang benar adalah: ${jawaban}.\n"
        tulis_bersih "Tekan ENTER untuk melanjutkan atau CTRL+C untuk keluar."

        baca_input huruf

        kembalikan_nilai_awal
        kursor_ke_atas 14
        teruskan
    akhir_jika

    # IFS digunakan agar baca_input a.k.a read tidak melakukan trimming whitespace
    # lihat https://unix.stackexchange.com/a/18889
    selama IFS= baca_input -n1 huruf; lakukan
        break;
    selesai

    jika [ "$huruf" = "" ]; maka
        kursor_ke_atas 13 && teruskan
    akhir_jika

    jika [ "$huruf" = " " ]; maka
        kursor_ke_atas 12 && teruskan
    akhir_jika

    jika [ "$( adakah_huruf_dalam "$jawaban" "$huruf" )" = "ya"  ]; maka
        # masukkan huruf ke tampungan jawaban
        jika [ "$huruf_benar" = "" ]; maka
            huruf_benar="$huruf"
        akhir_jika

        jika [ "$( adakah_huruf_dalam "$huruf_benar" "$huruf" )" = "tidak" ]; maka
            huruf_benar="${huruf_benar}${huruf}"
        akhir_jika

        jawaban_pengguna="$( tulis_garis_bawah "$jawaban" "kosong" )"
        kursor_ke_atas 12
        teruskan
    akhir_jika

    percobaan=$(( $percobaan + 1 ))

    kursor_ke_atas 12
selesai

samples/piramida.sh
#!/bin/bash
#
# Contoh file penggunaan alias bash-in-bahasa.

# Perintah ini wajib disertakan agar alias dikenali
shopt -s expand_aliases

# Asumsi anda menjalankan sample script ini dari root direktori bash-in-bahasa
. ./bash-in-bahasa.sh

tulis 'Piramida 1'
tulis '=========='
untuk (( baris=1; baris<=5; baris++ ))
lakukan

    untuk (( kolom=1; kolom<=$baris; kolom++ ))
    lakukan
        tulis -n '* '
    selesai

    tulis
selesai

tulis
tulis 'Piramida 2'
tulis '=========='
baris=5
selama (( $baris >= 1 ))
lakukan
    kolom=$baris
    selama (( $kolom >= 1 ))
    lakukan
        tulis -n '* '
        kolom=$(( $kolom - 1 ))
    selesai

    baris=$(( $baris - 1 ))
    tulis
selesai

tulis
tulis 'Piramida 3'
tulis '=========='

tulis -n 'Masukkan jumlah baris: '
read jumlah_baris

untuk (( baris=1; baris<=$jumlah_baris; baris++ ))
lakukan
    # Tulis spasi untuk padding kiri agar piramida dapat terlihat ketengah
    untuk (( spasi=1; spasi<=$(( $jumlah_baris - $baris )); spasi++ ))
    lakukan
        tulis -n ' '
    selesai

    untuk (( kolom=1; kolom<=$baris; kolom++ ))
    lakukan
        tulis -n '* '
    selesai

    tulis
selesai


samples/tebak-angka.sh
#!/bin/bash
#
# Contoh file penggunaan alias bash-in-bahasa.

# Perintah ini wajib disertakan agar alias dikenali
shopt -s expand_aliases

# Asumsi anda menjalankan sample script ini dari root direktori bash-in-bahasa
# $ bash samples/tebak-angka.sh
. ./bash-in-bahasa.sh

maksimal_percobaan=3
angka_minimum=1
angka_maksimum=15
jawaban_pengguna="salah"
percobaan=1
# Lihat https://stackoverflow.com/questions/8988824/generating-random-number-between-1-and-10-in-bash-shell-script
jawaban=$(( ($RANDOM % $angka_maksimum) + 1 ))

tulis '**********************************'
tulis 'SELAMAT DATANG DI GAME TEBAK ANGKA'
tulis '**********************************'
tulis "Kamu harus menebak angka yang muncul maksimal ${maksimal_percobaan} kali percobaan.
Angka berada diantara ${angka_minimum} dan ${angka_maksimum}. Selamat bermain!"

tulis
tulis -n '> Masukkan nama kamu: '
baca_input nama_pengguna

tulis

selama (( $percobaan <= $maksimal_percobaan ))
lakukan
    tulis -n "> #${percobaan} Masukkan angka: "
    baca_input jawaban_pengguna

    jika [ $jawaban = $jawaban_pengguna ];
    maka
        tulis "Selamat ${nama_pengguna}, jawaban kamu BENAR!"
        keluar 0
    akhir_jika

    percobaan=$(( $percobaan + 1 ))
selesai

tulis "GAME OVER. Ooops kamu sudah melakukan ${maksimal_percobaan} kali percobaan."
keluar 1